src/arrayMethodJoin.js

```diff
@@ -5,7 +5,40 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let str = '';
+    const arr = [...this];
+    const chengeddArr = [];
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === undefined || this[i] === null) {
+        arr[i] = '';
+      }
+    }
+
+    if (!this.length) {
+      return '';
+    }
```

Comment:
Looks too complicated, actually we only need two if's to check current element for null and undefined and check current index

```
      if (this[i] !== null && this[i] !== undefined) {
        str += this[i];
      }

      if (i !== this.length - 1) {
        str += separator;
      }
```